dev/integration_tests/semantics/test_driver/main_test.dart

```diff
@@ -0,0 +1,230 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+
+import 'android_testing.dart';
+
+void main() {
+  group('semantics suite', () {
+    FlutterDriver driver;
+
+    Future<AndroidSemanticsNode> getSemantics(SerializableFinder finder) async {
+      final int id = await driver.getSemanticsId(finder);
+      final String data = await driver.requestData('getSemanticsNode#$id');
+      return new AndroidSemanticsNode.deserialize(data);
+    }
+
+    setUpAll(() async {
+      driver = await FlutterDriver.connect();
+      await driver.setSemantics(true);
+    });
+
+    group('SelectionControls', () {
+      setUpAll(() async {
+        await driver.tap(find.text('SelectionControls'));
+      });
+
+      test('Checkbox has correct Android semantics', () async {
+        const String checkboxKey = 'SelectionControls#Checkbox1';
```

Comment:
Maybe place these as consts in a seperate dart file that can be imported by the app under test and the test driver, see for example https://github.com/flutter/flutter/blob/409007827bfc1f91d02415b3cc7243ad9c85f2fd/dev/integration_tests/ui/lib/keys.dart.